airbyte-cdk/python/airbyte_cdk/sources/utils/record_helper.py

```diff
@@ -20,8 +20,8 @@ def stream_data_to_airbyte_message(
     if schema is None:
         schema = {}
 
-    if isinstance(data_or_message, dict):
-        data = data_or_message
+    if isinstance(data_or_message, Mapping):
```

Comment:
`StreamData` is being defined as `Union[Mapping[str, Any], AirbyteMessage]` but it is not true that `dict == Mapping`. I fear there might be other cases where we have this issue